conan/internal/api/select_pattern.py

```diff
@@ -15,11 +16,19 @@ def split(s, c, default=None):
             return tokens[0], default
 
         recipe, package = split(expression, ":")
+        self.raw = expression
         self.ref, rrev = split(recipe, "#", rrev)
+        self.name, self.version = split(self.ref, "/")
```

Comment:
user and channel? Maybe testing is necessary with refs like ``pkg/0.1@user/channel``?